src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -332,6 +332,120 @@ static async ValueTask<int> FinishReadAsync(Task<int> readTask, byte[] localBuff
+        /// <paramref name="offset"/> and <paramref name="count"/> exceed the length of <paramref name="buffer"/>.
+        /// </exception>
+        /// <exception cref="EndOfStreamException">
+        /// Thrown when the end of the stream is reached before reading <paramref name="count"/> number of bytes.
+        /// </exception>
+        public ValueTask ReadExactlyAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken = default)
+        {
+            ValidateReadExactlyBufferArguments(buffer, offset, count);
+
+            ValueTask<int> vt = ReadAtLeastAsyncCore(buffer.AsMemory(offset, count), count, throwOnEndOfStream: true, cancellationToken);
+
+            // transfer the ValueTask<int> to a ValueTask without allocating here.
+            return ValueTask.DangerousCreateFromTypedValueTask(vt);
+        }
+
+        /// <summary>
+        /// Asynchronously reads at least a minimum number of bytes from the current stream, advances the position within the stream by the
+        /// number of bytes read, and monitors cancellation requests.
+        /// </summary>
+        /// <param name="buffer">The region of memory to write the data into.</param>
+        /// <param name="minimumBytes">The minimum number of bytes to read into the buffer.</param>
+        /// <param name="throwOnEndOfStream">
+        /// <see langword="true"/> to throw an exception if the end of the stream is reached before reading <paramref name="minimumBytes"/> of bytes;
+        /// <see langword="false"/> to return less than <paramref name="minimumBytes"/> when the end of the stream is reached.
+        /// The default is <see langword="true"/>.
+        /// </param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>
+        /// A task that represents the asynchronous read operation. The value of its Result property contains the total number of bytes read into
+        /// the buffer. This is guaranteed to be greater than or equal to <paramref name="minimumBytes"/> when <paramref name="throwOnEndOfStream"/>
+        /// is <see langword="true"/>. This will be less than <paramref name="minimumBytes"/> when the end of the stream is reached and
+        /// <paramref name="throwOnEndOfStream"/> is <see langword="false"/>. This can be less than the number of bytes allocated in the buffer if
+        /// that many bytes are not currently available.
+        /// </returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="minimumBytes"/> was not a positive value, or greater than the length of <paramref name="buffer"/>.
+        /// </exception>
+        /// <exception cref="EndOfStreamException">
+        /// Thrown when <paramref name="throwOnEndOfStream"/> is <see langword="true"/> and the end of the stream is reached before reading
+        /// <paramref name="minimumBytes"/> of data.
```

Comment:
```suggestion
        /// <paramref name="throwOnEndOfStream"/> is <see langword="true"/> and the end of the stream is reached before reading
        /// <paramref name="minimumBytes"/> bytes of data.
```

Reply:
Does `minimumBytes bytes of data.` make sense with the redundancy?